src/api-service/__app__/onefuzzlib/azure/nsg.py

```diff
@@ -0,0 +1,484 @@
+#!/usr/bin/env python
+#
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+import logging
+import os
+from typing import Dict, List, Optional, Set, Union, cast
+
+from azure.core.exceptions import HttpResponseError, ResourceNotFoundError
+from azure.mgmt.network.models import (
+    NetworkInterface,
+    NetworkSecurityGroup,
+    SecurityRule,
+    SecurityRuleAccess,
+    Subnet,
+    VirtualNetwork,
+)
+from msrestazure.azure_exceptions import CloudError
+from onefuzztypes.enums import ErrorCode
+from onefuzztypes.models import Error, NetworkSecurityGroupConfig
+from onefuzztypes.primitives import Region
+from pydantic import BaseModel, validator
+
+from .creds import get_base_resource_group
+from .network_mgmt_client import get_network_client
+
+
+def is_concurrent_request_error(err: str) -> bool:
+    return "The request failed due to conflict with a concurrent request" in str(err)
+
+
+def get_nsg(name: str) -> Optional[NetworkSecurityGroup]:
+    resource_group = get_base_resource_group()
+
+    logging.debug("getting nsg: %s", name)
+    network_client = get_network_client()
+    try:
+        nsg = network_client.network_security_groups.get(resource_group, name)
+        return cast(NetworkSecurityGroup, nsg)
+    except (ResourceNotFoundError, CloudError) as err:
+        logging.debug("nsg %s does not exist: %s", name, err)
```

Comment:
Should this be at a higher log level?

Reply:
Yup, fixing this one

Reply:
There are several other places with this pattern, that I didn't comment on. Might want to check all your new `except` blocks.

Reply:
Done, checked them all. The ones where we are "ignoring" the exceptions, I kept those as debug, since other existing components log as debug in the same situation